roles/openshift_health_checker/openshift_checks/docker_image_availability.py

```diff
@@ -0,0 +1,82 @@
+# pylint: disable=missing-docstring
+from openshift_checks import OpenShiftCheck, get_var
+from openshift_checks.mixins import NotContainerizedMixin
+
+
+class DockerImageAvailability(NotContainerizedMixin, OpenShiftCheck):
+    """Check that required Docker images are available."""
+
+    name = "docker_image_availability"
+    tags = ["preflight"]
+
+    def run(self, tmp, task_vars):
+        openshift_release = get_var(task_vars, "openshift_release")
+
+        # attempt to fetch fully qualified image tag
+        openshift_facts = self.module_executor("openshift_facts", {}, tmp, task_vars)
+        openshift_image_tag = self.get_image_tag(openshift_facts)
+
+        failed_pulls = set()
+
+        # attempt to pull each image and fail on
+        # the first one that cannot be fetched
+        pulled_images = self.pull_images(self.rpm_docker_images(), openshift_release, tmp, task_vars)
+        if "failed" in pulled_images and "failed_images" in pulled_images:
+            failed_pulls.update(pulled_images["failed_images"])
+
+        if not openshift_image_tag:
+            openshift_image_tag = get_var(task_vars, "openshift_image_tag")
+            if not openshift_image_tag:
+                return {"failed": True, "msg": "Unable to retrieve fully qualified image tag. Failed to fetch images: %s" % (set(failed_pulls).update(self.rpm_qualified_docker_images()))}
+
+        # pull each required image that requires a qualified image
+        # tag (v0.0.0.0) rather than a standard release format tag (v0.0)
+        pulled_qualified_images = self.pull_images(self.rpm_qualified_docker_images(), "v" + openshift_image_tag, tmp, task_vars)
+        if "failed" in pulled_qualified_images and "failed_images" in pulled_qualified_images:
+            failed_pulls.update(pulled_qualified_images["failed_images"])
+
+        if failed_pulls:
+            return {"failed": True, "msg": "Failed to pull required docker images: %s" % failed_pulls}
+
+        return {"changed": True}
+
+    @staticmethod
+    def rpm_docker_images():
+        return [
+            "openshift3/registry-console",
+        ]
+
+    @staticmethod
+    def rpm_qualified_docker_images():
+        return [
+            "openshift3/ose-haproxy-router",
+            "openshift3/ose-docker-registry",
+            "openshift3/ose-deployer",
+            "openshift3/ose-pod",
+        ]
+
+    def pull_images(self, images, tag, tmp, task_vars):
+        failed = set()
+        for image in images:
+            args = {
+                "name": image + ":" + tag
+            }
+
+            res = self.module_executor("docker_image", args, tmp, task_vars)
+            if "failed" in res:
```

Comment:
Subtle bug: `"failed"` might be in the result, but it could be `failed: False`...